### DISCUSSION TITLE
timeline variables with nested objects

### DISCUSSION THREAD

**Discussion by CrumpLab** [jspsych/jsPsych#650] [Q&A]
I've been trying to use timeline variables that contain nested objects. For example, I modified this from the example documentation on timeline variables. Here, I would be attempting to populate the `stimulus` say with the info contained in the `props.type` part of the timeline array.

But, this doesn't seem to work. I'm probably missing something like a syntax issue here. Is there a workaround?

```
var face_name_procedure = {
    timeline: [
        {
            type: 'html-keyboard-response',
            stimulus: jsPsych.timelineVariable('props.type'),
            choices: jsPsych.NO_KEYS,
            trial_duration: 2500
        }
    ],
    timeline_variables: [
        { face: 'person-1.jpg' , props: {type: 'A', condition: 'A'}},
        { face: 'person-2.jpg' , props: {type: 'B', condition: 'A'}},
        { face: 'person-3.jpg' , props: {type: 'C', condition: 'B'}},
        { face: 'person-4.jpg' , props: {type: 'D', condition: 'B'}}
    ]
}
```

**Comment by jodeleeuw** [comment]
The string that is passed to `jsPsych.timelineVariable()` is used to [match keys in the object](https://github.com/jspsych/jsPsych/blob/master/jspsych.js#L550-L556). `props` is a valid key, but `props.type` is not since you would need to extract the `props` object and get its `type` key.

You can workaround by customizing how you invoke the `timelineVariable` function:

```js
{
  stimulus: function() { 
    var props = jsPsych.timelineVariable('props', true); // true argument means evaluate immediately
    return props.type;
  }
}
```

**Comment by CrumpLab** [comment]
Woohoo!! Thanks for the quick reply, my lack of js basics was driving me nuts.
